errors.go

```diff
@@ -215,6 +235,41 @@ func (errorType ErrorType) UserErrorf(msg string, args ...interface{}) error {
 	}
 }
 
+// DetailsAdd adds a details element to an error
+func (errorType ErrorType) DetailsAdd(err error, details interface{}) error {
+	if err == nil {
+		return errorType.Details(details)
+	}
+
+	c := new(customError)
+	c.error = errors.WithStack(err)
+	c.userMessage = GetUserMessage(err)
+
+	origDetails := GetDetails(err)
+	if origDetails != nil {
+		c.details = append(origDetails, details)
+	} else {
+		c.details = []interface{}{details}
+	}
+
+	if errorType != NoType {
+		c.errorType = errorType
+	} else {
+		c.errorType = GetType(err)
+	}
+
+	return c
+}
+
+// Details creates a new error with arbitrary details
+func (errorType ErrorType) Details(details interface{}) error {
```

Comment:
I don`t see an actual usage for that.
Sadly, I think we would always have to use 2 lines if we want to add details.

```
err = weberr.Unauthorized.Wrapf(err, "user is unauthorized")
err = weberr.AddDetails(err, detailsObj)
```

Do you have a use case where this func is useful for?

Reply:
I don't have user-facing use case, no. But the `AddDetails` function relies on it as a safeguard if someone calls it with a `nil` error object (just like `UserWrapf` does).

In short, `Details` is to `AddDetails` what `UserErrorf` is to `UserWrapf`.

Reply:
I think that if someone called AddDetails on a nil error maybe we should return a nil error.
Seems like a misuse to me.
`error()` function on that object would return `""`, `%v` also. I don't know what's the best way to handle that, but I don't think creating empty error with details is a good practice